client/client.go

```diff
@@ -245,8 +245,13 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 			continue
 		}
 		if resp.StatusCode/100 == 5 {
-			// TODO: make sure this is a no leader response
-			cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s has no leader", eps[k].String()))
+			switch resp.StatusCode {
+			case http.StatusInternalServerError:
+				// TODO: make sure this is a no leader response
+				cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s has no leader", eps[k].String()))
+			default:
+				cerr.Errors = append(cerr.Errors, fmt.Errorf("client: etcd member %s returns server error [%s] with body [%s]", eps[k].String(), http.StatusText(resp.StatusCode), body))
```

Comment:
1. I do no think we should return the full body.
2. We should not retry in some of the cases. If you want to fully resolve the TODO, you need to think carefully about this.